### PR TITLE
fix(app): Fix Run Completion and RunSummary malfunction

### PR DESCRIPTION
**Pull request by koji**
# Overview
- Call `stopRun` when tapping Run Completed/Run Failed button to fix 👇
https://github.com/Opentrons/opentrons/pull/12597#pullrequestreview-1407720327
- Call reset to clone run from the `runId`
- Fix ConfirmCancelRunModal (call `dismissCurrentRun` after `stopRun`)
- Add style to protocol name to clamp a long name.

# Test Plan
- send a protocol to a robot/dev kit (if it doesn't have any protocol)
- tap `All protocols`
- run a protocol
- tap RunCompleted/RunFailed button
Open Desktop app and there is no `Go to Run` button in the robot card
- tap Run again in RunSummary
Reload Check Desktop app and there is `Go to Run` button in the robot card
- tap cancel button and tap cancel in the modal
ODD app shows RunSummary
Reload Desktop app and there is no `Go to Run` button in the robot card

# Changelog
- call dismissCurrentRun from ConfirmRunCancelModal to cancel a run properly
- call closeCurrentRun when tapping RunCompleted / RunCancel button
- activate trackProtocolEvent and switch from hard-code to const 

# Review requests

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] fix run cancel button 
- [x] fix RunCompleted / RunFailed button

File: app/src/organisms/OnDeviceDisplay/RunningProtocol/ConfirmCancelRunModal.tsx [hunk 1]
```diff
@@ -99,7 +108,7 @@ export function ConfirmCancelRunModal({
             buttonType="alert"
             buttonText={t('cancel_run')}
             onClick={handleCancelRun}
-            disabled={isCanceling}
+            disabled={isCanceling && isDismissing}
```
Comment (b-cooper) [inline app/src/organisms/OnDeviceDisplay/RunningProtocol/ConfirmCancelRunModal.tsx]: Should this be an `||` ?

**Review by b-cooper — APPROVED** [review]
Code looks good, one small comment. 👍

**Comment by ewagoner** [comment]
What ticket is this for?

**Comment by ewagoner** [comment]
https://user-images.githubusercontent.com/2960/236250229-f7baeacd-e86b-4eaa-8cae-840ea005a246.mov

I'm seeing a full-screen error modal on this branch when I cancel run setup. Is that intentional?

**Comment by koji** [comment]
> What ticket is this for?

RCORE-750

**Comment by koji** [comment]
@ewagoner 
thank you for pointing out that.
`Cancel run` from Prepare to Run should redirect to `ProtocolDetails` (checked this to Mel)
I will update.

**Pull request merged**